src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.cpp

```diff
@@ -167,52 +172,130 @@ tnsr::ijj<DataType, SpatialDim, Frame> deriv_spatial_metric(
 }
 
 template <size_t SpatialDim, typename Frame, typename DataType>
-tnsr::ii<DataType, SpatialDim, Frame> time_deriv_of_spatial_metric(
+void time_deriv_of_spatial_metric(
+    const gsl::not_null<tnsr::ii<DataType, SpatialDim, Frame>*>
+        dt_spatial_metric,
     const Scalar<DataType>& lapse,
     const tnsr::I<DataType, SpatialDim, Frame>& shift,
     const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
     const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept {
-  auto dt_spatial_metric =
-    make_with_value<tnsr::ii<DataType, SpatialDim, Frame>>(get(lapse), 0.);
+  if (get_size(get<0, 0>(*dt_spatial_metric)) != get_size(get(lapse))) {
+    tnsr::ii<DataType, SpatialDim, Frame> dt_spatial_metric_(get(lapse));
+    *dt_spatial_metric = dt_spatial_metric_;
+  }
+  auto& d0_spatial_metric = *dt_spatial_metric;
```

Comment:
This notation is odd.  This alias saves you all of two characters later in the function, so how about just dropping it?